src/javaserver/Requests/RequestHandler.java

```diff
@@ -0,0 +1,79 @@
+package javaserver.Requests;
+
+import javaserver.Routes.Route;
+import javaserver.Routes.RoutesRegistrar;
+
+public class RequestHandler {
+
+    public static final String OK = "200 OK";
+    public static final String NOT_FOUND = "404 Not Found";
+    public static final String UNAUTHORIZED = "401 Unauthorized";
+
+    private final RoutesRegistrar routes = RoutesRegistrar.getInstance();
+
+    private HttpRequestParser parser;
+    private Logger logger;
+
+    public RequestHandler(HttpRequestParser parser) {
+        this.parser = parser;
+        this.logger = new Logger();
+    }
+
+    public RequestHandler(HttpRequestParser parser, Logger logger) {
+        this.parser = parser;
+        this.logger = logger;
+    }
+
+    public String status() {
+        if(containsRoute()) {
+            if(routes.isSecured(parser.uri())) {
+                if(parser.containsHeader("Authorization")) {
```

Comment:
This is a lot of levels of nesting. I would first try to convert this to a single level of `if`/`else if`, and then we can make further improvements from there.